common/persistence/visibility_triple_manager.go

```diff
@@ -36,15 +36,15 @@ import (
 
 type (
 	visibilityTripleManager struct {
-		logger                    log.Logger
-		dbVisibilityManager       VisibilityManager
-		pinotVisibilityManager    VisibilityManager
-		esVisibilityManager       VisibilityManager
-		readModeIsFromPinot       dynamicconfig.BoolPropertyFnWithDomainFilter
-		readModeIsFromES          dynamicconfig.BoolPropertyFnWithDomainFilter
-		writeMode                 dynamicconfig.StringPropertyFn
-		logCustomerQueryParameter dynamicconfig.BoolPropertyFnWithDomainFilter
-		readModeIsDouble          dynamicconfig.BoolPropertyFnWithDomainFilter
+		logger                     log.Logger
+		dbVisibilityManager        VisibilityManager
+		primaryVisibilityManager   VisibilityManager
+		secondaryVisibilityManager VisibilityManager
+		readModeIsFromPrimary      dynamicconfig.BoolPropertyFnWithDomainFilter
+		readModeIsFromSecondary    dynamicconfig.BoolPropertyFnWithDomainFilter
```

Comment:
side comment: Having these two as separate configs is confusing but we need 3 states so a single config wouldn't suffice. This should have been a single property of type enum (NoShadowing, SecondaryIsShadow, PrimaryIsShadow)

Reply:
Yeah, I am changing the name to make it more generic so it can be reused in OS migration, but these names make the logic harder to understand now. Maybe it is better to call source and destination?

Reply:
As for these 2 properties, they are actually enableReadVisibilityFromES and enableReadVisibilityFromPinot, 
we have separate fliprs to control each domain's read source. And one it finished migration and switches to  visibility dual manager, we can use the same flipr property for that visibility store

Reply:
Updated the names!